packages/graphiql-react/src/editor/tabs.ts

```diff
@@ -278,10 +280,12 @@ export function useSetEditorValues({
     }) => {
       queryEditor?.setValue(query ?? '');
       variableEditor?.setValue(variables ?? '');
-      headerEditor?.setValue(headers ?? '');
+      if (typeof headers === 'string' || typeof defaultHeaders === 'string') {
+        headerEditor?.setValue(headers ?? defaultHeaders ?? '');
```

Comment:
```suggestion
        headerEditor?.setValue(headers ?? defaultHeaders);
```
not needed either `headers` or `defaultHeaders` will be strings